src/MagnumPlugins/TinyGltfImporter/TinyGltfImporter.cpp

```diff
@@ -414,53 +426,62 @@ std::unique_ptr<AbstractMaterialData> TinyGltfImporter::doMaterial(const Unsigne
             flags |= PhongMaterialData::Flag::DiffuseTexture;
         }
-    }
 
-    auto specularIt = material.extPBRValues.find("specularShininessTexture");
-    if(specularIt != material.extPBRValues.end()) {
-        specularTexture = specularIt->second.TextureIndex();
-        flags |= PhongMaterialData::Flag::SpecularTexture;
-    }
+        auto specularTextureValue = cmnBlinnPhongExt.Get("specularShininessTexture");
+        if(specularTextureValue.Type() != tinygltf::NULL_TYPE) {
+            specularTexture = UnsignedInt(specularTextureValue.Get("index").Get<int>());
+            flags |= PhongMaterialData::Flag::SpecularTexture;
+        }
 
-    /* Colors */
-    Vector3 diffuseColor{1.0f};
-    auto diffuseFactorIt = material.extPBRValues.find("diffuseFactor");
-    if(diffuseFactorIt != material.extPBRValues.end()) {
-        const tinygltf::ColorValue& diffuseRaw = diffuseFactorIt->second.ColorFactor();
-        diffuseColor = Vector3{float(diffuseRaw[0]), float(diffuseRaw[1]), float(diffuseRaw[2])};
-    }
+        /* Colors */
+        auto diffuseFactorValue = cmnBlinnPhongExt.Get("diffuseFactor");
+        if(diffuseFactorValue.Type() != tinygltf::NULL_TYPE) {
+            diffuseColor = Vector4{Vector4d{
+                diffuseFactorValue.Get(0).Get<double>(),
+                diffuseFactorValue.Get(1).Get<double>(),
+                diffuseFactorValue.Get(2).Get<double>(),
+                diffuseFactorValue.Get(3).Get<double>()}};
+        }
 
-    Vector3 specularColor{0.0f};
-    auto specularFactorIt = material.extPBRValues.find("specularFactor");
-    if(specularFactorIt != material.extPBRValues.end()) {
-        const tinygltf::ColorValue& specularRaw = specularFactorIt->second.ColorFactor();
-        specularColor = Vector3{float(specularRaw[0]), float(specularRaw[1]), float(specularRaw[2])};
-    }
+        auto specularColorValue = cmnBlinnPhongExt.Get("specularFactor");
+        if(specularColorValue.Type() != tinygltf::NULL_TYPE) {
+            specularColor = Vector3{Vector3d{
+                specularColorValue.Get(0).Get<double>(),
+                specularColorValue.Get(1).Get<double>(),
+                specularColorValue.Get(2).Get<double>()}};
+        }
 
-    /* Parameters */
-    Float shininess{1.0f};
-    auto shininessFactorIt = material.extPBRValues.find("shininessFactor");
-    if(shininessFactorIt != material.extPBRValues.end()) {
-        shininess = shininessFactorIt->second.Factor();
+        /* Parameters */
+        auto shininessFactorValue = cmnBlinnPhongExt.Get("shininessFactor");
+        if(shininessFactorValue.Type() != tinygltf::NULL_TYPE) {
+            shininess = float(shininessFactorValue.Get<double>());
+        }
+    } else {
+        auto dt = material.values.find("baseColorTexture");
+        if(dt != material.values.end()) {
+            diffuseTexture = dt->second.TextureIndex();
+            flags |= PhongMaterialData::Flag::DiffuseTexture;
+        }
```

Comment:
Similarly to `baseColorTexture`, there's the `baseColorFactor`. As I mentioned in #42, the glTF sample model repository uses it on even the most basic models. Assimp imports it as diffuse, so we could do the same here (unless `diffuseFactor` is set).